### PR TITLE
[15.6] Fixes VSTS 581661: [Feedback] /* */ Comments slow or non responsive

### PR DESCRIPTION
**Pull request by mkrueger**
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/581661

Seems that a span change detection was missing completely - that this
worked in former versions must've been a side effect. I took the
chance to ensure that highlighting is really done in background which
looks ok on my systems.

### CODE REVIEW

**Comment by mkrueger** [comment]
y could repro that - seems it's updating the first time.

**Comment by neerajdeshmukh** [comment]
We have verified bug on Visual Studio Community 2017 for Mac Version 7.4.1 (build 33).

Issue is still reproducible. Also code commenting via (/*..*/) is not working properly.
Issues getting like code is not getting commented completely. Also ({) coding is not getting updated.
After un-commenting, code is not getting un-commented. 

Screencast:
https://www.screencast.com/t/torbfEjMkRLn

Build Taken from PR:
https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1482035&tab=artifacts&_a=summary

Build Gist:
https://gist.github.com/neerajdeshmukh/61584e0ca3e2fa9225ee4d97efbe6f3b

Hence marking as QA-Failed. Reopening related bug https://devdiv.visualstudio.com/DevDiv/_workitems/edit/581661

Kindly let us know if anything else is required from our end.

**Comment by neerajdeshmukh** [comment]
Just to update, issue is still reproducible on new build provided 7.4.1.34
1. Build Gist: https://gist.github.com/neerajdeshmukh/07be3cdde6a55644362ab9bc0217f4cf
2. Screencast: https://www.screencast.com/t/tuD3tof1J
3. Build Taken from: https://devdiv.visualstudio.com/DevDiv/_build/indexbuildId=1485111&tab=artifacts&_a=summary

**Comment by neerajdeshmukh** [comment]
Verified on Visual Studio Community 2017 for Mac Version 7.4.1 (build 35). Issue is still reproducible.

Build Taken from: https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1486187&tab=artifacts&_a=summary
Screencast: https://www.screencast.com/t/H34qGgxKYs
Build Gist: https://gist.github.com/neerajdeshmukh/c534884a1f6ecf94474e4d1dfc212998

File: main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextViewMargin.cs [hunk 1]
```diff
@@ -1322,12 +1321,28 @@ internal void DisposeLayoutDict ()
...
 		public void PurgeLayoutCache ()
 		{
 			DisposeLayoutDict ();
 		}
 
+		void PurgeLayoutCacheAfter (int lineNumber)
+		{
+			foreach (var descr in layoutDict.ToArray()) {
+				if (descr.Key >= lineNumber) {
+					Console.WriteLine ("purge " + descr.Key);
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextViewMargin.cs]: Debug code.

**Comment by neerajdeshmukh** [comment]
Verified on Visual Studio Community 2017 for Mac Version 7.4.1 (build 37).
It is working fine and no delay observed while commenting code. Hence marking as QA-Approved.

Screencast: https://www.screencast.com/t/OhLVZe7K4

Build Gist: https://gist.github.com/neerajdeshmukh/d3a7ee120fc9c3ce6a515c1502cdd8a0

Build Taken from:
https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1492012&tab=artifacts&_a=summary

Approved for Merge.

**Pull request merged**